Assets/Scripts/PlayerMovement.cs

```diff
@@ -36,7 +39,35 @@ void Update()
             rb.velocity = Vector3.zero;
         }
 
-        transform.Translate(new Vector3(horizontal, 0, vertical) * (speed * Time.deltaTime));
+        // THIS BIT IS THE PROBLEM
+        // if (rb.velocity != Vector3.zero)
+        // {
+        //     transform.rotation = Quaternion.RotateTowards(transform.rotation, 
+        //     Quaternion.LookRotation(new Vector3(transform.GetComponent<Rigidbody>().velocity.x, 0, transform.GetComponent<Rigidbody>().velocity.z)), 
+        //     Time.deltaTime * RotateSpeed);
+        // }
+
+        transform.Translate(new Vector3(-1 * vertical, 0, horizontal) * (speed * Time.deltaTime));
+    }
+
+    void FixedUpdate()
```

Comment:
What this?

Reply:
That's the raycast solution to the wall problem

the method name:

void update() happens once per frame
void fixedUpdate() happens once per delta time, so if you're running at 1000fps this doesn't update a million times, and happens in time with the physics engine

Reply:
ooooo that's cool